packages/server/src/routes/tenants.js

```diff
@@ -17,6 +17,40 @@ router.get('/', requireUser, async (req, res) => {
     res.json(response);
 });
 
+router.post('/', async (req, res) => {
+    if (!req.body.tenant_name || !req.body.agency_name || !req.body.agency_abbreviation) {
+        res.status(400).send('Tenant name, agency name, and agency abbreivation is required');
+        return;
+    }
+
+    try {
+        const agency = {
+            name: req.body.agency_name,
+            abbreviation: req.body.agency_abbreviation,
+            parent: 0,
+            warning_threshold: 30,
+            danger_threshhold: 15,
+        };
+
+        const createdAgency = await createAgency(agency, 15);
```

Comment:
Hey @bspates, I had a quick question for you - I'm using the updated `createAgency` endpoint but I'm unable to successfully create an agency and the creation hangs. I was able to create before the change - but as far as I can tell I'm using this correctly. Would you have an idea on what I'm doing wrong here?

Reply:
@slants3 Do you get any error? What is the hardcoded number 15? Shouldn't you be using the user id that is making the request?

Reply:
@slants3 Also I see the tenant creation is happening after this. You'll want to swap those around. How the foreign key dependencies in the tables work now is that only main_agency_id in tenant can be left null to start. So you have to create the tenant, then the agency, then go back and update the main_agency_id in the tenant.

Reply:
>Shouldn't you be using the user id that is making the request?

👍, should use `req.session.user.id` (available when using `requireAdminUser` as suggested in a separate comment)

>So you have to create the tenant, then the agency, then go back and update the main_agency_id in the tenant.

👍, this should also mean `setAgencyTenantId` is not necessary since agencies must be created with a `tenant_id` to start with and cannot move between tenants.